course/src/main/java/org/openedx/course/presentation/calendarsync/CalendarManager.kt

```diff
@@ -0,0 +1,396 @@
+        val cursor = calendarContentResolver.query(
+            CalendarContract.Calendars.CONTENT_URI, projection,
+            CalendarContract.Calendars.ACCOUNT_NAME + "=? and (" +
+                    CalendarContract.Calendars.NAME + "=? or " +
+                    CalendarContract.Calendars.CALENDAR_DISPLAY_NAME + "=?)", arrayOf(
+                accountName, calendarTitle,
+                calendarTitle
+            ), null
+        )
+        if (cursor?.moveToFirst() == true) {
+            if (cursor.getString(cursor.getColumnIndex(CalendarContract.Calendars.NAME))
+                    .equals(calendarTitle)
+            ) {
+                calendarId =
+                    cursor.getInt(cursor.getColumnIndex(CalendarContract.Calendars._ID)).toLong()
+            }
+        }
+        cursor?.close()
+        return calendarId
+    }
+
+    /**
+     * Method to add important dates of course as calendar event into calendar of mobile app
+     */
+    fun addEventsIntoCalendar(
+        calendarId: Long,
+        courseId: String,
+        courseName: String,
+        courseDateBlock: CourseDateBlock
+    ) {
+        val date = courseDateBlock.date.toCalendar()
+        // start time of the event, adjusted 1 hour earlier for a 1-hour duration
+        val startMillis: Long = date.timeInMillis - TimeUnit.HOURS.toMillis(1)
+        // end time of the event added to the calendar
+        val endMillis: Long = date.timeInMillis
+
+        val values = ContentValues().apply {
+            put(CalendarContract.Events.DTSTART, startMillis)
+            put(CalendarContract.Events.DTEND, endMillis)
+            put(
+                CalendarContract.Events.TITLE,
+                "${resourceManager.getString(R.string.course_assignment_due_tag)} : $courseName"
+            )
+            put(
+                CalendarContract.Events.DESCRIPTION,
+                getEventDescription(
+                    courseId = courseId,
+                    courseDateBlock = courseDateBlock,
+                    isDeeplinkEnabled = corePreferences.appConfig.courseDatesCalendarSync.isDeepLinkEnabled
+                )
+            )
+            put(CalendarContract.Events.CALENDAR_ID, calendarId)
+            put(CalendarContract.Events.EVENT_TIMEZONE, TimeZone.getDefault().id)
+        }
+        val uri = context.contentResolver.insert(CalendarContract.Events.CONTENT_URI, values)
+        uri?.let { addReminderToEvent(uri = it) }
+    }
+
+    /**
+     * Method to generate & add deeplink into event description
+     *
+     * @return event description with deeplink for assignment block else block title
+     */
+    private fun getEventDescription(
+        courseId: String,
+        courseDateBlock: CourseDateBlock,
+        isDeeplinkEnabled: Boolean
+    ): String {
+        val eventDescription = courseDateBlock.title
+//        if (isDeeplinkEnabled && !TextUtils.isEmpty(courseDateBlock.blockId)) {
```

Comment:
if it is commented for a reason this should be documented

Reply:
but why did we commented this, maybe we can make isDeeplinkEnabled: Boolean = false by default?

Reply:
The commented code is related to Branch SDK which is yet to be implemented. Uncommenting the code will result in a compile error.